### PR TITLE
Chk-2126: payment method with clientId

### PR DESCRIPTION
**Pull request by infantesimone**
depends on https://github.com/pagopa/pagopa-ecommerce-local/pull/108

#### List of Changes

- add `x-client-id` header param for:
  - `GET payment-methods`
  - `GET payment-methods/:id`
- add `clientId` as body param in new payment methods
- filter all get methods with clientId
- updata mongo document with clientId
- new compound unique index for payment method, in only to have unique payment method given clientID, name and type code
#### Motivation and Context

The goal of this PR is to handle differente payment methods according to client ID, so touchpoint Id related to payment methods.

#### How Has This Been Tested?

- run unit test
- run integration test

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by Fernando-Granato — COMMENTED** [review]
Please fix CR collection tests into ecommerce-local project CD collection tests into pagopa-checkout-tests project

File: src/test/java/it/pagopa/ecommerce/payment/methods/controller/PaymentMethodsControllerTests.java [hunk 1]
```diff
@@ -49,14 +49,14 @@ class PaymentMethodsControllerTests {
     private Tracer tracer;
 
     @Test
-    void shouldCreateNewmethod() {
```
Comment (Fernando-Granato) [inline src/test/java/it/pagopa/ecommerce/payment/methods/controller/PaymentMethodsControllerTests.java:52]: You can duplicate this test for client id checkout and IO

File: src/test/java/it/pagopa/ecommerce/payment/methods/controller/PaymentMethodsControllerTests.java [hunk 2]
```diff
@@ -49,14 +49,14 @@ class PaymentMethodsControllerTests {
...
-        PaymentMethodRequestDto paymentMethodRequestDto = TestUtil.getPaymentMethodRequest();
+    void shouldCreateNewMethodForCheckout() {
+        PaymentMethodRequestDto paymentMethodRequestDto = TestUtil.getPaymentMethodRequestForCheckout();
 
         PaymentMethod paymentMethod = TestUtil.getPaymentMethod();
 
         PaymentMethodResponseDto methodResponse = TestUtil.getPaymentMethodResponse(paymentMethod);
 
-        Mockito.when(paymentMethodService.createPaymentMethod(any(), any(), any(), any(), any()))
+        Mockito.when(paymentMethodService.createPaymentMethod(any(), any(), any(), any(), any(), any()))
```
Comment (Fernando-Granato) [inline src/test/java/it/pagopa/ecommerce/payment/methods/controller/PaymentMethodsControllerTests.java:59]: use eq(ClientID.CHECKOUT)

File: src/test/java/it/pagopa/ecommerce/payment/methods/controller/PaymentMethodsControllerTests.java [hunk 3]
```diff
@@ -71,11 +71,36 @@ void shouldCreateNewmethod() {
...
-    void shouldGetAllMethods() {
+    void shouldCreateNewMethodForIO() {
+        PaymentMethodRequestDto paymentMethodRequestDto = TestUtil.getPaymentMethodRequestForIO();
 
         PaymentMethod paymentMethod = TestUtil.getPaymentMethod();
 
-        Mockito.when(paymentMethodService.retrievePaymentMethods((int) TestUtil.getTestAmount())).thenReturn(
+        PaymentMethodResponseDto methodResponse = TestUtil.getPaymentMethodResponse(paymentMethod);
+
+        Mockito.when(paymentMethodService.createPaymentMethod(any(), any(), any(), any(), any(), any()))
```
Comment (Fernando-Granato) [inline src/test/java/it/pagopa/ecommerce/payment/methods/controller/PaymentMethodsControllerTests.java:81]:
```suggestion
        Mockito.when(paymentMethodService.createPaymentMethod(any(), any(), any(), any(), any(), eq(ClientID.IO)))
```

File: src/test/java/it/pagopa/ecommerce/payment/methods/controller/PaymentMethodsControllerTests.java (hunk 2 above)
Comment (Fernando-Granato) [inline src/test/java/it/pagopa/ecommerce/payment/methods/controller/PaymentMethodsControllerTests.java:59]:
```suggestion
        Mockito.when(paymentMethodService.createPaymentMethod(any(), any(), any(), any(), any(), eq(CLientID.CHECKOUT)))
```

File: src/main/java/it/pagopa/ecommerce/payment/methods/infrastructure/PaymentMethodDocument.java [hunk 4]
```diff
@@ -3,6 +3,8 @@
 import lombok.AllArgsConstructor;
 import lombok.Data;
 import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.index.CompoundIndex;
+import org.springframework.data.mongodb.core.index.CompoundIndexes;
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/ecommerce/payment/methods/infrastructure/PaymentMethodDocument.java:7]: unused imports?

File: src/main/java/it/pagopa/ecommerce/payment/methods/infrastructure/PaymentMethodRepository.java [hunk 5]
```diff
@@ -14,4 +14,17 @@ Mono<PaymentMethodDocument> findByPaymentMethodNameOrPaymentMethodTypeCode(
 
     Flux<PaymentMethodDocument> findByPaymentMethodStatus(String paymentMethodStatus);
 
+    Flux<PaymentMethodDocument> findByClientId(String clientId);
+
+    Mono<PaymentMethodDocument> findByPaymentMethodIDAndClientId(
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/ecommerce/payment/methods/infrastructure/PaymentMethodRepository.java:19]:
```suggestion
    Mono<PaymentMethodDocument> findByPaymentMethodIdAndClientId(
```
🚲

**Review by pietro-tota — APPROVED** [review]
Well done! just some comments🚀

File: src/main/java/it/pagopa/ecommerce/payment/methods/infrastructure/PaymentMethodDocument.java (hunk 4 above)
Comment (infantesimone) [reply src/main/java/it/pagopa/ecommerce/payment/methods/infrastructure/PaymentMethodDocument.java:7]: 👀

Comment (infantesimone) [reply src/main/java/it/pagopa/ecommerce/payment/methods/infrastructure/PaymentMethodDocument.java]: done

**Review by pietro-tota — APPROVED** [review]

**Comment by infantesimone** [comment]
https://dev.azure.com/pagopaspa/pagoPA-projects/_build/results?buildId=129037&view=logs&j=5a5405a0-8439-550a-2a82-0e4c8988d6a1&t=5a5405a0-8439-550a-2a82-0e4c8988d6a1 integration tests

**Pull request merged**